mlir/tools/mlir-miopen-driver/mlir-miopen-driver.cpp

```diff
@@ -1269,27 +1291,16 @@ static LogicalResult populateValidationLogic(
         builder.create<AllocOp>(builder.getUnknownLoc(), verifyMemRefType);
     block->push_back(verifyHostAllocOp);
 
-    auto filterMemRefForBf16Type = MemRefType::get(
-        ArrayRef<int64_t>(filterDimension.begin(), filterDimension.end()),
-        floatType);
-    auto inputMemRefForBf16Type = MemRefType::get(
-        ArrayRef<int64_t>(inputDimension.begin(), inputDimension.end()),
-        floatType);
-    auto outputMemRefForBf16Type = MemRefType::get(
-        ArrayRef<int64_t>(outputDimension.begin(), outputDimension.end()),
-        floatType);
-
     auto unknownSizeMemRefFloatType =
         MemRefType::get({-1, -1, -1, -1}, floatType);
 
     auto verifyUnkownSizeMemRefCastOp = builder.create<MemRefCastOp>(
         builder.getUnknownLoc(), verifyHostAllocOp, unknownSizeMemRefFloatType);
     block->push_back(verifyUnkownSizeMemRefCastOp);
 
-    auto cpuMemConvertOp = FuncOp::create(
-        builder.getUnknownLoc(), "mcpuMemBF16ConvertFloat",
-        builder.getFunctionType(
-            {fourDimUnknownSizeMemRefType, unknownSizeMemRefFloatType}, {}));
+    auto cpuMemConvertOp = makeFuncDecl(
```

Comment:
is this `cpuMemConvertOp` really needed for all data types? line 1099 seems to have it already for bf16(i16) path.